### DISCUSSION TITLE
A way to detect drag and drop to a different parent node

### DISCUSSION THREAD

**Discussion by ZakKa89** [xyflow/xyflow#3253] [New Features]
#  A way to detect drag and drop to a different parent node

ttps://reactflow.dev/docs/guides/sub-flows/

With subflows, you can add nodes to parent nodes. However, I'd like the ability to drag and drop a child node from one parent node to another. The nodes array will only update the positioning, but not detect that you've put it in a different parent node. 

Ideally, the node would 'snap' so it would fully position itself inside the parent node, and can't occupy space between parent nodes. 

## Why is this useful for me?

I'm using parent nodes as a workaround for swimlanes and I'd like the user to be able to move a node to a different swimlane. 

I could do some calculations and detect by the x-axis positioning in which parent node the node is, but I would be nice if it's supported by this framework

<img width="707" alt="image" src="https://github.com/wbkd/react-flow/assets/24491661/963a8fe1-73aa-4657-b31f-e8ae225e43b9">

**Comment by bcakmakoglu** [comment]
You can check for intersections of nodes using `getIntersectingNodes` and find out if a node is inside another parent or not.

[There's a basic example](https://reactflow.dev/docs/examples/nodes/intersections/), though it doesn't do "dynamic" grouping or anything.

Other than that, there's always `document.elementsFromPoint(x, y)` which you can use to find all elements from a given x and y position (not the graph x, y though) and finding out if an element with the class `parent` is present on given coordinates.

**Comment by ZakKa89** [comment]
Sounds like that's exactly what I need thanks!
